libraries/Microsoft.Bot.Builder/Integration/BotFrameworkOptions.cs

```diff
@@ -43,6 +44,13 @@ public BotFrameworkOptions()
         /// <seealso cref="BotAdapter.Use(IMiddleware)"/>
         public IList<IMiddleware> Middleware { get; } = new List<IMiddleware>();
 
+        /// <summary>
+        /// Gets a list of the <see cref="BotState"/> providers to use on each incoming activity.
```

Comment:
>each incoming activity. [](start = 74, length = 23)

Is the implication that there is always a BotStateSet() added which consumes this?  And only in Asp.NET integrations?

Reply:
BotStateSet() actually doesn't consume this.  We may have to revisit this.

The BotFrameworkOptions.State list was created as a mechanism to enable other Singletons to consume to create State Accessors.  In the example LuisBot (see other PR), we created a Singleton which reads the options.State list and creates all the accessors that will be consumed in the IBot-derived class (which is transient).  It's a bit complicated, but i'll add more details on it's intended usage.

Note: Yochay yesterday was questioning this relationship as well.  Hmm..

Reply:
Here's where I landed.  

Gets a list of the <see cref="BotState"/> providers to use on each incoming activity.  Objects in the State list enable other components to get access to the state providers during the start up process.  For example, creating state property accessors within a ASP.net Core Singleton that could be passed to your IBot-derived class.
The providers in this list are not associated with the BotStateSet Middleware component. To clarify, state providers in this list are not automatically loaded or saved during the turn process.